### ISSUE TITLE
Explaining trained model using kaggle x-ray dataset

### ISSUE THREAD

**Issue by GoSXiang** [albermax/innvestigate#167]
![image](https://user-images.githubusercontent.com/54198222/67137835-e6db4e80-f26d-11e9-92df-d813f3c2fa22.png)

Hi, I am using a model I trained using keras convolutional layers for the prediction of NORMAL/PNEUMONIA images. The analyzer used here is the deep taylor method. A sample image is taken from the true negative class (NORMAL) and 
it seems that the model is using information from the 'shoulders' to actually predict that the lungs are normal? The model that I trained has an 81.25% accuracy on the 16 test images. What could be wrong here?

**Comment by heytitle** [comment]
Interesting case. imho, it's quite difficult to justify what went wrong here by looking at one sample.

Would it be possible for you to provide more samples (possibly come from the same labels) and their explanations?

**Comment by GoSXiang** [comment]
![image](https://user-images.githubusercontent.com/54198222/67172917-c935f280-f3ef-11e9-8381-77e875c7906e.png)

![image](https://user-images.githubusercontent.com/54198222/67172981-e23ea380-f3ef-11e9-96d3-2f00774971a0.png)

![image](https://user-images.githubusercontent.com/54198222/67173018-fedadb80-f3ef-11e9-9da7-ecb1836f6b4d.png)

These are the cases of true negative classes. By right, information from the shoulders shouldnt have been used for the prediction of 'NORMAL' class right? How is that so, given that the model has a decent accuracy 81.25% on 16 images aka 13 out of 16 images correct? The code is attached below:


import imp
import innvestigate
import innvestigate.utils as iutils

imgnetutils = imp.load_source("utils_imagenet", "/content/drive/My Drive/src/utils_imagenet.py")

# Define methods
methods = [
    # NAME                    OPT.PARAMS                POSTPROC FXN                TITLE
    # Function
    ("gradient",              {"postprocess": "abs"},   imgnetutils.graymap,       "Gradient"),
    
    ("smoothgrad",            {"postprocess": "square"},imgnetutils.graymap,       "SmoothGrad"),

    ("lrp.sequential_preset_a_flat",{"epsilon": 1},     imgnetutils.heatmap,       "LRP-PresetAFlat")
]

# Create model without trailing softmax
model_wo_softmax = iutils.keras.graph.model_wo_softmax(model)

# Create analyzers.
analyzers = []
for method in methods:
  analyzer = innvestigate.create_analyzer(method[0],        # analysis method identifier
                                          model_wo_softmax, # model without softmax output
                                          **method[1])      # optional analysis parameters
  analyzers.append(analyzer)


analyzer_names = ['Gradient', 'Smooth Gradient', 'LRP Preset A Flat']

# First true negative sample
first_sample = tn[:,3,:,:,:]

counter = 1
plt.figure(figsize=(17.5,17.5))

plt.subplot(1,4,counter)
plt.xlabel('Original Image')
plt.imshow(first_sample.squeeze())


for i,analyzer in enumerate(analyzers):
  a = analyzer.analyze(first_sample)
  a = imgnetutils.postprocess(a, "BGRtoRGB",channels_first=False)
  a = methods[i][2](a)
  
  counter += 1
  
  plt.subplot(1,4,counter)
  plt.xlabel(analyzer_names[i])
  plt.imshow(a[0],interpolation='none')
